components/membership/synchronisation-impl/src/integrationTest/kotlin/net/corda/membership/impl/synchronisation/MemberSynchronisationIntegrationTest.kt

```diff
@@ -0,0 +1,378 @@
+            }
+        }
+
+        fun setupConfig() {
+            val publisher = publisherFactory.createPublisher(PublisherConfig("clientId"), bootConfig)
+            publisher.publish(
+                listOf(
+                    Record(
+                        CONFIG_TOPIC,
+                        MESSAGING_CONFIG,
+                        Configuration(messagingConf, messagingConf, 0, schemaVersion)
+                    )
+                )
+            )
+            configurationReadService.start()
+            configurationReadService.bootstrapConfig(bootConfig)
+        }
+
+        @JvmStatic
+        @AfterAll
+        fun tearDown() {
+            configurationReadService.stop()
+        }
+    }
+
+    @Test
+    fun `member updates are are successfully received on sync message from MGM`() {
+        groupPolicyProvider.putGroupPolicy(TestGroupPolicy())
+
+        // Create membership package to be published
+        val members: List<MemberInfo> = mutableListOf(createTestMemberInfo(participant))
+        val dummySignature = CryptoSignatureWithKey(
+            ByteBuffer.wrap(source.x500Name.toByteArray()),
+            ByteBuffer.wrap(source.x500Name.toByteArray()),
+            KeyValuePairList(
+                listOf(
+                    KeyValuePair("name", participant.x500Name)
+                )
+            )
+        )
+        val signedMembers = members.map {
+            SignedMemberInfo.newBuilder()
+                .setMemberContext(ByteBuffer.wrap(keyValueSerializer.serialize(it.memberProvidedContext.toAvro())))
+                .setMgmContext(ByteBuffer.wrap(keyValueSerializer.serialize(it.mgmProvidedContext.toAvro())))
+                .setMemberSignature(dummySignature)
+                .setMgmSignature(dummySignature)
+                .build()
+        }
+        val membership = SignedMemberships.newBuilder()
+            .setMemberships(signedMembers)
+            .setHashCheck(SecureHash("SHA-256", ByteBuffer.wrap("1234567890".toByteArray())))
+            .build()
+
+        val membershipPackage = MembershipPackage.newBuilder()
+            .setDistributionType(DistributionType.STANDARD)
+            .setCurrentPage(0)
+            .setPageCount(1)
+            .setCpiWhitelist(null)
+            .setGroupParameters(null)
+            .setMemberships(
+                membership
+            )
+            .setDistributionMetaData(
+                DistributionMetaData(
+                    "id",
+                    clock.instant(),
+                )
+            )
+            .build()
+
+        // Start subscription to gather results of processing synchronisation command
+        val completableResult = CompletableFuture<Pair<String, PersistentMemberInfo>>()
+        val registrationRequestSubscription = subscriptionFactory.createPubSubSubscription(
+            SubscriptionConfig("membership_updates_test_receiver", MEMBER_LIST_TOPIC),
+            getTestProcessor { s, e ->
```

Comment:
Nit picky suggestion. `s, e` were used previously to capture states and events, but in yours it's key and value so maybe `k, v` would be better.
```suggestion
            getTestProcessor { k, v ->
```

Reply:
I added one commnet below about possibly changing the test processor signature. If doing that, this could change to just `getTestProcessor { v ->`

Reply:
Done